src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -133,6 +138,31 @@ bool has_f16_constants(const std::shared_ptr<const ov::Model>& function) {
     return false;
 }
 
+bool is_large_language_model(const std::shared_ptr<const ov::Model>& model) {
+    using namespace ov::pass::pattern;
+
+    const auto past = wrap_type<ov::op::v6::ReadValue>();
+    const auto convert_past = wrap_type<ov::op::v0::Convert>({past});
```

Comment:
minor:
it's better to use 
pattern::optional it will work similar to Or but will simplify the pattern

rv = wrap_type<ReadValue>
convert = optional<Convert>(rv)
gather = wrap_type<Gather>

ReadValue -> Gather
ReadValue -> Convert -> Gather will be matched

Reply:
Implemented, thanks!